core/commands/files.go

```diff
@@ -147,9 +148,18 @@ var filesStatCmd = &cmds.Command{
 			dagserv = node.DAG
 		}
 
-		nd, err := getNodeFromPath(req.Context, node, api, path)
-		if err != nil {
-			return err
+		var nd ipld.Node
+		if strings.HasPrefix(path, "/ipfs/") {
+			nd, err = getNodeFromDag(req.Context, api, path)
+			if err != nil {
+				return err
+			}
+		} else {
+			nd, err = getNodeFromMfs(node, path)
+			if err != nil {
```

Comment:
This will return *any* error form MFS, ideally we should check for errors like `os.ErrNotExist` like in

https://github.com/ipfs/go-unixfs/blob/c54d0e47ad42f03de7736c2826c315de9f63b5a6/io/directory.go#L187-L188

Unfortunately I don't think this is guaranteed in the API at the moment, meaning that `BasicDirectory` does return that error (although I'm not sure if it does in all the appropriate cases) and  `HAMTDirectory` definitively doesn't.

So, let's merge this as is, but please add a big `TODO:` here requesting a stable error from the MFS API.

Reply:
https://github.com/ipfs/go-unixfs/blob/c54d0e47ad42f03de7736c2826c315de9f63b5a6/hamt/hamt.go#L396

It seems that `HAMTDirectory` also returns `os.ErrNotExist`. Is it better to add `os.ErrNotExist` check?

Reply:
Good catch! So, I'm not sure we're returning that error in all the appropriate circumstances, it depends on your main objective, if you want to err on the side of returning maybe more than real "file doesn't exist" (`os.ErrNotExist`) errors or in avoiding the 500 internal error when we actually  should just have returned the `os.ErrNotExist` error, both options are fine by me.